src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardSyntaxClassifier.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification.Classifiers
+{
+    internal class DiscardSyntaxClassifier : AbstractSyntaxClassifier
+    {
+        public override ImmutableArray<Type> SyntaxNodeTypes { get; } = ImmutableArray.Create(
+            typeof(DiscardDesignationSyntax),
+            typeof(DiscardPatternSyntax),
+            typeof(ParameterSyntax),
+            typeof(IdentifierNameSyntax));
+
+        public override void AddClassifications(
+           Workspace workspace,
+           SyntaxNode syntax,
+           SemanticModel semanticModel,
+           ArrayBuilder<ClassifiedSpan> result,
+           CancellationToken cancellationToken)
+        {
+            if (syntax.IsKind(SyntaxKind.DiscardDesignation) || syntax.IsKind(SyntaxKind.DiscardPattern))
+            {
+                result.Add(new ClassifiedSpan(syntax.Span, ClassificationTypeNames.Keyword));
+                return;
+            }
+
+            switch (syntax)
+            {
+                case ParameterSyntax parameter:
+                    var symbol = semanticModel.GetDeclaredSymbol(parameter, cancellationToken);
```

Comment:
@lameox Thanks for sticking with this. Please add a check that `parameter.Identifier` is `"_"`.

Reply:
consider writing thins as:

```c#
            switch (syntax)
            {
                case ParameterSyntax parameter when parameter.Identifier.Text == "_": ...
                case IdentifierName name when name.Identifier.Text == "_": ...
             }
```

(note: no default as fallthrough is fine here).

Reply:
done. 

out of curiosity: the check on the `parameter.Identifier.Text` should be redundant in all cases where `IParameterSymbol.IsDiscard` is true right? is this just a precaution in case other syntax might be introduced in the future that leads to `IParameterSymbol.IsDiscard` here? and wouldn't we then want that new syntax to be classified as a discard and thus highlighted?

Reply:
Getting SymbolInfo from the SemanticModel is work. Avoiding work when classifying gives a better experience to the user.

Reply:
Ahh ok so the same as with the identifier names. Thanks for the reply :)

Reply:
👍

Reply:
in general: semantic/symbolic work is much mre expensive than syntactic work.  so if we can do a synactic check that allows us to avoid the semantic work most of the time, we should do that.